ocs_ci/helpers/managed_services.py

```diff
@@ -0,0 +1,191 @@
+"""
+Managed Services related functionalities
+"""
+import logging
+import yaml
+
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.node import (
+    get_osd_running_nodes,
+    get_worker_nodes,
+    get_node_objs,
+    get_node_pods,
+)
+from ocs_ci.ocs.resources.pod import get_ceph_tools_pod, get_osd_pods
+from ocs_ci.ocs.resources.pvc import get_all_pvc_objs
+from ocs_ci.utility.utils import convert_device_size, run_cmd
+
+log = logging.getLogger(__name__)
+
+
+def verify_provider_topology():
+    """
+    Verify topology in a Managed Services provider cluster
+
+    1. Verify replica count
+    2. Verify total size
+    3. Verify OSD size
+    4. Verify OSD running worker nodes count
+    5. Verify worker node instance type
+    6. Verify OSD count
+    7. Verify OSD cpu
+    8. Verify machine pools
+    9. Verify OSD running nodes are part of the correct machinepool
+    10. Verify that other pods are not running on OSD nodes
+
+    """
+    # importing here to avoid circular import
+    from ocs_ci.ocs.resources.storage_cluster import StorageCluster, get_osd_count
+
+    size = f"{config.ENV_DATA.get('size', 4)}Ti"
+    replica_count = 3
+    osd_size = 4
+    instance_type = "m5.xlarge"
+    size_map = {
+        "4Ti": {"total_size": 12, "osd_count": 3, "instance_count": 3},
+        "8Ti": {"total_size": 24, "osd_count": 6, "instance_count": 3},
+        "12Ti": {"total_size": 36, "osd_count": 9, "instance_count": 6},
+        "16Ti": {"total_size": 48, "osd_count": 12, "instance_count": 6},
+        "20Ti": {"total_size": 60, "osd_count": 15, "instance_count": 9},
+        "48Ti": {"total_size": 144, "osd_count": 36, "instance_count": 18},
+        "96Ti": {"total_size": 288, "osd_count": 72, "instance_count": 36},
+    }
+    cluster_namespace = constants.OPENSHIFT_STORAGE_NAMESPACE
+    storage_cluster = StorageCluster(
+        resource_name="ocs-storagecluster",
+        namespace=cluster_namespace,
+    )
+
+    # Verify replica count
+    assert (
+        int(storage_cluster.data["spec"]["storageDeviceSets"][0]["replica"])
+        == replica_count
+    ), "Replica count is not as expected"
```

Comment:
I would add the replica count and expected replica count to the assert message here (so it will be directly visible what exactly is the problem).

Reply:
Noted. This will be added in the next PR.